js/boaexchange.js

```diff
@@ -0,0 +1,1021 @@
+        await this.loadMarkets ();
+        let response = undefined;
+        try {
+            let request = { 'orderId': id };
+            response = await this.privateGetOrdersOrderId (this.extend (request, params));
+        } catch (e) {
+            throw e;
+        }
+        if (!response['data']) {
+            throw new OrderNotFound (this.id + ' order ' + id + ' not found');
+        }
+        return this.parseOrder (response['data']);
+    }
+
+    async fetchOrders (symbol = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        let request = { 'market': symbol, 'begin': since, 'limit': limit, 'status': 'init,open,closed,cancelled' };
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['market'] = market['id'];
+        }
+        let response = await this.privateGetOrders (this.extend (request, params));
+        let orders = this.parseOrders (response['data'], market, since, limit);
+        return this.filterBySymbol (orders, symbol);
+    }
+
+    async fetchOrderBook (symbol, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        let type = 'both';
+        let response = await this.publicGetMarketsLabelOrderbook (this.extend ({
+            'label': this.marketId (symbol),
+            'side': type,
+            'limit': limit,
+        }, params));
+        let orderbook = response['data'];
+        return this.parseOrderBook (orderbook, undefined, 'buy', 'sell', 0, 1);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let response = await this.publicGetMarketsLabel (this.extend ({
+            'label': this.marketId (symbol),
+        }, params));
+        let ticker = response['data'];
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        let response = await this.publicGetMarkets (this.extend ({
+        }, params));
+        let tickers = response['data'];
+        let result = {};
+        for (let t = 0; t < tickers.length; t++) {
+            let ticker = tickers[t];
+            let id = ticker['id'];
+            let market = undefined;
+            let symbol = undefined;
+            if (id in this.markets_by_id) {
+                market = this.markets_by_id[id];
+                if (market) {
+                    symbol = market['symbol'];
+                }
+            } else {
+                symbol = this.parseSymbol (symbol);
+                market = this.markets (symbol);
+            }
+            result[symbol] = this.parseTicker (ticker, market);
+        }
+        return result;
+    }
+
+    async fetchTradingFees (params = {}) {
+        await this.loadMarkets ();
+        let market_key = Object.keys (this.markets)[0];
```

Comment:
This looks like a small bug to me, because we always get the same market, not the one we're asking for. For that reason we have `fetchTradingFee (symbol, params = {})`. So, it has to be rewired a bit.

Reply:
Either way the fee is the same across all pairs on the exchange. So the call still works as it should

Reply:
For now... but we don't want it to suddenly break down if that changes or the exchange defines a new fee schedule, right? (That has happenned to us several times already ;))

Reply:
Okay. let me speak to the exchange and i'll get them to make a modification to it now

Reply:
Modification made and pushed